### PR TITLE
Dockerfile dependencies cleanup

### PR DESCRIPTION
**Pull request by rowanruseler**
Dockerfile:
  included,
    - apt-get clean command, clears out the local repository of
    downloaded package files
    - remove packages list since no longer needed after apt-get install

.gitignore, added Vim love

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,18 @@
...
+[._]s[a-v][a-z]
+[._]sw[a-p]
+# session
+Session.vim
+# temporary
+.netrwhist
+*~
+# auto-generated tag files
+tags
+/jooq/events/src/main/gen/
```
Comment (timfennis) [inline .gitignore]: Aren't these supposed to go in your global gitignore.

Comment (timfennis) [reply .gitignore]: Talking about the vim and /jooq stuff

Comment (timfennis) [reply .gitignore]: https://git-scm.com/docs/gitignore

Comment (rowanruseler) [reply .gitignore]: Not everybody uses a global gitignore; but good point, will removed.

**Pull request merged**